plugins/API/API.php

```diff
@@ -129,6 +129,14 @@ public function getSegmentsMetadata($idSites = array(), $_hideImplementationData
                     $segment->setPermission($isAuthenticatedWithViewAccess);
                 }
 
+                if ($segment->getSqlSegment() && $segment->getUnionOfSegments()) {
+                    throw new \Exception(sprintf('Union of segments and SQL segment is set for segment "%s", use only one of them', $segment->getName()));
+                }
+
+                if ($segment->getUnionOfSegments() && in_array($segment->getSegment(), $segment->getUnionOfSegments(), true)) {
+                    throw new \Exception(sprintf('The segment %s contains a union segment to itself', $segment->getName()));
+                }
```

Comment:
Did you think about throwing these exceptions in `core\Plugin\Segment.php`? It seems to me this validation would fit better there.